front-end-side/src/Pages/EnterData.js

```diff
@@ -8,23 +8,21 @@ const ShowData = () => {
     const [date, setDate] = useState('');
     const [seller, setSeller] = useState('');
     const [amount, setAmount] = useState('');
-    const [info, setInfo] = useState([]);
     const [details, setDetails] = useState('');
-    const [status, setStatus] = useState(false);
     const postData = async (e) => {
         e.preventDefault();
-        const res = await axios.post('https://localhost:7174/api/ShowData', { "date": date, "seller": seller, "amount": amount, "details": details });
-        navigate('/data');
-    }
-    const handleSubmit = async(e) => {
-        e.preventDefault();
-        await axios.get('https://localhost:7174/api/ShowData').then(resp => {
-            setInfo(resp);
-            setStatus(true);
-        })
-    }
-    const show = () => {
-        setStatus(true);
+        const dateRegEx = /^([0-9]{4}[-/ ]?((0[13-9]|1[012])[-/ ]?(0[1-9]|[12][0-9]|30)|(0[13578]|1[02])[-/ ]?31|02[-/ ]?(0[1-9]|1[0-9]|2[0-8]))|([0-9]{2}(([2468][048]|[02468][48])|[13579][26])|([13579][26]|[02468][048]|0[0-9]|1[0-6])00)[-/ ]?02[-/ ]?29)$/
+        const amountRegEx = /^\d+\.?\d{1,2}$/
+        if (!dateRegEx.test(date)) {
+            alert("Date is invalid");
+        }
+        else if (!amountRegEx.test(amount)) {
+            alert("Amount is invalid");
```

Comment:
Can we show error not as an alert, but as text in our page?

Reply:
Done ✅